test/bench_suite.go

```diff
@@ -0,0 +1,150 @@
+package test
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"reflect"
+	"regexp"
+	"testing"
+)
+
+var allBenchmarksFilter = func(_, _ string) (bool, error) { return true, nil }
+var matchBenchMethod = flag.String("testify.b", "", "regular expression to select benchmarks of the testify suite to run")
+
+// Suite is a basic testing suite with methods for storing and
+// retrieving the current *testing.B context.
+type Suite struct {
+	b *testing.B
+}
+
+// B retrieves the current *testing.B context.
+func (suite *Suite) B() *testing.B {
+	return suite.b
+}
+
+// SetB sets the current *testing.B context.
+func (suite *Suite) SetB(b *testing.B) {
+	suite.b = b
+}
+
+// Run takes a testing suite and runs all of the benchmarks attached
+// to it.
+func Run(b *testing.B, suite BenchingSuite) {
+	suite.SetB(b)
```

Comment:
@xcoulon Yes, I tried to follow the same approach testify offers for our suite. I tried to do that to avoid missing different type of Suites which, at the end, will have conflicts. That is way I created a new suite which extends what testify has inside of their package